packages/medusa/src/api/routes/store/carts/create-cart.ts

```diff
@@ -222,6 +265,39 @@ export default async (req, res) => {
   res.status(200).json({ cart: cleanResponseData(cart, []) })
 }
 
+async function retrieveVariantsWithIsolatedProductModule(
+  remoteQuery,
+  variantIds: string[]
+) {
+  const variantIdsMap = new Map(variantIds.map((id) => [id, true]))
+
+  const variables = {
+    filters: {
+      variants: {
+        id: [...variantIds],
+      },
+    },
+  }
+
+  const query = {
+    product: {
+      __args: variables,
+      ...defaultStoreProductRemoteQueryObject,
```

Comment:
todo: why? here we just need some product/variant prop but not all, I suggest to have something related to this endpoint and the needs instead of fetching the entire relations tree. wdyt?

Reply:
Agree, changed to fetch only necessary things